OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardActivity.java

```diff
@@ -23,14 +31,48 @@ public static Intent newIntent(@NonNull Activity activity,
         return intent;
     }
 
+    private SnackbarErrorNotification snackbarErrorNotification;
+    private boolean isFullScreenErrorVisible = true;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         blockDrawerFromOpening();
+        snackbarErrorNotification = new SnackbarErrorNotification(findViewById(R.id.coordinator_layout));
     }
 
     @Override
     public Fragment getFirstFragment() {
         return CourseTabsDashboardFragment.newInstance();
     }
+
+    @Override
+    public void hideSnackBar() {
+        snackbarErrorNotification.hideError();
+    }
+
+    @Override
+    public void setSnackBarVisibility(boolean fullScreenErrorVisibility) {
+        this.isFullScreenErrorVisible = fullScreenErrorVisibility;
+        if (fullScreenErrorVisibility) {
+            snackbarErrorNotification.hideError();
+        } else if (!NetworkUtil.isConnected(this)) {
+            snackbarErrorNotification.showOfflineError(this);
+        }
+    }
+
+    @Override
+    public void onRestart() {
+        super.onRestart();
+        if (NetworkUtil.isConnected(this)) {
+            snackbarErrorNotification.hideError();
+        } else if (!isFullScreenErrorVisible) {
+            snackbarErrorNotification.showOfflineError(this);
+        }
+    }
+
+    @Override
+    public void onRefresh() {
+        EventBus.getDefault().post(new RefreshListener.RefreshEvent());
```

Comment:
**RefreshEvent**
It looks like a generic refresh event and seems all the clients who have registered it will take the callback. Possibly clients not only belongs to Course Dashboard. So we should make the scope of it limited to Course Dashboard only.